cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -82,7 +85,7 @@ public BasicMapReduceContext(Program program,
                                MetricsCollectionService metricsCollectionService,
                                DatasetFramework dsFramework,
                                CConfiguration conf) {
-    super(program, runId, runtimeArguments, datasets,
+    super(program, runId, extractScope(program.getName(), runtimeArguments), datasets,
```

Comment:
It is interesting that you are doing this inside the mapreduce, not in the workflow driver. I am undecided on what is better. In a way I feel that doing it in the workflow would be one central place to do it for all program types. I am not even sure whether the MapReduceContext is the very first thing that is constructed by the map reduce (and will that be the case for all program types?). It seems to me that it is cleaner if MapReduce does not need to extract the arguments for its scope, but they should be provided to MapReduce by whoever initiates it.